### PR TITLE
Enhanced debug explain

### PR DESCRIPTION
**Pull request by bdbaddog**
Enhanced --debug=explain output to break out sources, depends, implicit dependencies to make it easier to figure which dependencies changed.

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: test/explain/basic.py [hunk 1]
```diff
@@ -324,9 +324,15 @@ def WriteInitialTest( valueDict ) :
...
-               new: ['zzz', 'yyy', 'xxx', '%(python_sep)s']
+scons: rebuilding `file3' because:
+           the dependency order changed:
+           ->Sources
+           Old:xxx	New:zzz
+           Old:yyy	New:yyy
+           Old:zzz	New:xxx
+           ->Depends
+           ->Implicit
+           Old:/usr/bin/python	New:/usr/bin/python
```
Comment (mwichmann) [inline test/explain/basic.py]: That explicit string as a Python path won't work on Windows.

Comment (bdbaddog) [reply test/explain/basic.py]: What do you mean?  I'm just dumping the bimplicit's..

Comment (mwichmann) [reply test/explain/basic.py]: See the appveyor build:  /usr/bin/python is not what it returns:

```
38/300 (12.67%) c:\\python27\\python.exe -tt test\explain\basic.py
STDOUT =========================================================================
12c12
<            Old:/usr/bin/python	New:/usr/bin/python
---
>            Old:C:\python27\python.exe	New:C:\python27\python.exe
FAILED test of C:\projects\scons\src\script\scons.py
	at line 615 of C:\projects\scons\testing\framework\TestCommon.py (_complete)
```

Comment (bdbaddog) [reply test/explain/basic.py]: Fixed.

**Pull request merged**